megamek/src/megamek/client/ui/swing/MiniMap.java

```diff
@@ -1267,57 +1255,57 @@ private void zoomIn() {
-                minimized = false;
-                initializeMap();
-            } else {
-                if (x < 14) {
-                    if (GUIPreferences.getInstance().getMouseWheelZoomFlip()) {
-                        zoomOut();
-                    } else {
-                        zoomIn();
-                    }
-                } else if ((x < 28) && (zoom > 2)) {
-                    heightDisplayMode = ((++heightDisplayMode) > NBR_MODES) ? 0 : heightDisplayMode;
-                    initializeMap();
-                } else if (x > (getSize().width - 14)) {
-                    if (GUIPreferences.getInstance().getMouseWheelZoomFlip()) {
-                        zoomIn();
-                    } else {
-                        zoomOut();
-                    }
-                } else {
-                    // Minimize button
-                    heightBuffer = getSize().height;
-                    setSize(getSize().width, 14);
-                    mapImage = ImageUtil.createAcceleratedImage(Math.max(1, getSize().width), 14);
-                    minimized = true;
-                    initializeMap();
-                }  
-            }
-        } else if (bv != null) {
-            if ((x < DIALOG_MARGIN) || (x > (getSize().width - leftMargin))
+    private void processMouseDrag(int x, int y, int modifiers) {
+        if ((x < DIALOG_MARGIN) || (x > (getSize().width - leftMargin))
                 || (y < topMargin)
-                || (y > (getSize().height - topMargin - 14))) {
-                return;
+                || (y > (getSize().height - topMargin - buttonHeight))) {
+            return;
+        }
+
+        if ((modifiers & InputEvent.CTRL_DOWN_MASK) != 0) {
+            bv.checkLOS(translateCoords(x - leftMargin, y - topMargin));
+        } else {
+            if (!dragging) {
+                dragging = true;
+                setCursor(Cursor.getPredefinedCursor(Cursor.MOVE_CURSOR));
             }
-            if ((me.getModifiersEx() & InputEvent.CTRL_DOWN_MASK) != 0) {
-                bv.checkLOS(translateCoords(x - leftMargin, y - topMargin));
+            bv.centerOnPointRel(
+                    ((double)(x - leftMargin))/(double)((HEX_SIDE_BY_SIN30[zoom] + HEX_SIDE[zoom])*board.getWidth()),
+                    ((double)(y - topMargin))/(double)(2 * HEX_SIDE_BY_COS30[zoom]*board.getHeight()));
+            bv.stopSoftCentering();
+            repaint();
+        }
+    }
+
+    private void processMouseClick(int x, int y) {
+        if ((x < 0) || (y < 0) || (x >= getSize().width) || (y >= getSize().height)
+                || (y <= getSize().height - buttonHeight)) {
+            return;
+        }
+        if (minimized) {
+            setSize(getSize().width, heightBuffer);
+            mapImage = ImageUtil.createAcceleratedImage(getSize().width, heightBuffer);
+            minimized = false;
+            initializeMap();
+        } else {
+            if (x < 14) {
```

Comment:
14 AND 28 AND 2!

All joking aside, can you toss in a little comment explaining what these 14s and 28s are?